editor/modes/visual-editor/block-list.js

```diff
@@ -23,14 +24,56 @@ class VisualEditorBlockList extends wp.element.Component {
 		this.onSelectionStart = this.onSelectionStart.bind( this );
 		this.onSelectionChange = this.onSelectionChange.bind( this );
 		this.onSelectionEnd = this.onSelectionEnd.bind( this );
+		this.onPointerMove = throttle( this.onPointerMove.bind( this ), 250 );
+		// Browser does not fire `*move` event when the pointer position changes
+		// relative to the document, so fire it with the last known position.
+		this.onScroll = () => this.onPointerMove( { clientY: this.lastClientY } );
+		this.setBlockRef = this.setBlockRef.bind( this );
 
 		this.state = {
 			selectionAtStart: null,
 		};
+
+		this.coordMap = {};
+		this.coordMapKeys = [];
+		this.lastClientY = 0;
+		this.refs = {};
+	}
+
+	componentDidUpdate() {
+		// Clean up refs.
+		this.refs = pickBy( this.refs );
```

Comment:
I made a wrong assumption here. I didn't think it would pass through here. In that case it seems better to do:

```js
if ( ref === null ) {
	delete this.refs[ uid ];
} else {
	this.refs = {
		...this.refs,
		[ uid ]: ref,
	};
}
```

Do you know why it is possible to delete a property from an object that is a class property, but not add one?